prefect_dask/task_runners.py

```diff
@@ -268,7 +268,11 @@ def visit_fn(expr):
     async def wait(self, key: UUID, timeout: float = None) -> Optional[State]:
         future = self._get_dask_future(key)
         try:
-            return await future.result(timeout=timeout)
+            result = future.result(timeout=timeout)
+            if inspect.isawaitable(result):
+                return await result
+            else:
+                return result
```

Comment:
No this isn't safe. We must be awaiting the retrieval of a result from a future. If it's not returning a coroutine we're blocking the event loop.

Reply:
Gotcha, reverted this change.